### DISCUSSION TITLE
Classify_poi : Only spatial polygons are supported

### DISCUSSION THREAD

**Discussion by MaximeAuffret** [r-lidar/lidR#630] [Q&A]
Hello, 
I use lidR to produce DEM from lidar data.
I recently updated all my packages and now the function `classify_poi`, that I use to classify the buildings, doesn't work anymore. here's what I get :

![image](https://user-images.githubusercontent.com/116190503/196740776-b3d87d74-e7c3-4263-a7e8-2ef12b946960.png)

Usually it worked without any problem. 
I tried to change the chunk size, the output DEM resolution, the `sf/sfc` or `sp` format of the building file.
Do you know where it come from ?

Usually it worked without any problem. 
I tried to change the chunk size, the output DEM resolution, the sf/sfc or sp format of the building file.
Do you know where it come from ?

```r
# The library that I use
library(lidR)
library(knitr)
library(shiny)
library(manipulateWidget)
library(raster)
library(ggplot2)
library(sf)

# The script
BAT <- st_read(dir_bat, quiet = T)    # Building file (.shp)
ZI_LAS = readLascatalog(x)              # Las file from IGN (Test urban area of Manosque (04) in France) the file is declassified for tests

mycfs <- csf(
    sloop_smooth      <- FALSE,
    class_threshold   <- 0.05,
    cloth_resolution  <- 0.5,
    rigidness         <- 1,
    iterations        <- 500,
    time_step         <- 0.65
  ) 

opt_laz_compression(ZI_LAS) = T
opt_output_files(ZI_LAS) <- paste0(dir_sol,"/Sol_{XCENTER}_{YCENTER}_{ID}")

ZI_SOL = classify_ground(ZI_LAS,algorithm = mycfs)
  
opt_chunk_buffer(ZI_SOL) <- 20
opt_chunk_size(ZI_SOL)   <- 250
plot(ZI_SOL, chunk = TRUE)

dir.create(paste0(dir_sol,"/BAT"))
dir_bat = paste0(dir_sol,"/BAT")

opt_output_files(ZI_SOL) <- paste0(dir_bat,"/Bat_{XCENTER}_{YCENTER}_{ID}")

st_crs(BAT) <- st_crs(ZI_SOL)
ZI_BAT <- classify_poi(ZI_SOL, class = as.integer(6), roi = BAT )
```

UPDATE : I use my scipt on two computer, it work in one of them so I think the problem comes from a package but i don't find wich one.

**Comment by Jean-Romain** [comment]
Please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) in the form of text copy pasted ([no image](https://meta.stackoverflow.com/questions/285551/why-should-i-not-upload-images-of-code-data-errors-when-asking-a-question)).

**Comment by Jean-Romain** [comment]
I guess a minimal reproducible example would be

```r

library(lidR)
library(sf)

x = "..."

BAT <- st_read(dir_bat, quiet = T)    # Building file (.shp)
ZI_SOL = readLAScatalog(x)            # Las file from IGN (Test urban area of Manosque (04) in France) the file is declassified for tests

opt_output_files(ZI_SOL) <- paste0(tmpdir(),"/Bat_{XCENTER}_{YCENTER}_{ID}")

st_crs(BAT) <- st_crs(ZI_SOL)
ZI_BAT <- classify_poi(ZI_SOL, class = as.integer(6), roi = BAT )
```
Now I need some data to reproduce.

**Comment by MaximeAuffret** [comment]
You can download the data here : https://wxs.ign.fr/mct8y415ma63eudk50qaf2bi/telechargement/prepackage/JEU-TEST_LIDARHD_PACK_17-05-2022$LIDARHD_1-0_HYB_JEU-TEST-DALLE-0923_6308-2021/file/LIDARHD_1-0_HYB_JEU-TEST-DALLE-0923_6308-2021.7z

The building file is attched here.
[BAT.zip](https://github.com/r-lidar/lidR/files/9827538/BAT.zip)

I removed the classification by using the extent (raster package) of the data and classify_poi (before the update).

**Comment by Jean-Romain** [comment]
The following worked fine. Again please provide a minimal reproducible code. Do not let met guess what code I should run to reproduce.

```r
library(lidR)
library(sf)

x = "lidR issues/issue 630/LIDARHD_1-0_HYB_JEU-TEST-DALLE-0923_6308-2021/"
dir_bat = "lidR issues/issue 630/BAT"

BAT <- st_read(dir_bat, quiet = T)    # Building file (.shp)
ZI_SOL = readLAScatalog(x)            # Las file from IGN (Test urban area of Manosque (04) in France) the file is declassified for tests

opt_chunk_buffer(ZI_SOL) <- 20
opt_chunk_size(ZI_SOL)   <- 250
opt_laz_compression(ZI_LAS) = T
opt_output_files(ZI_SOL) <- paste0(tempdir(),"/Sol_{XCENTER}_{YCENTER}_{ID}")

plot(ZI_SOL, chunk = TRUE)
plot(BAT, add = T)

st_crs(BAT) <- st_crs(ZI_SOL)
ZI_BAT <- classify_poi(ZI_SOL, class = as.integer(6), roi = BAT )
```

**Reply by MaximeAuffret** [reply]
Yes I know, I'm using it to test my reclassification process and see if I'm far from what the IGN (National Institute of Geographic and Forestry Information) can do.

**Reply by MaximeAuffret** [reply]
I'm sorry, you will find a link to download the data unclassified that I use. 
https://we.tl/t-FfwbczKHfc

I used this script to "unclassify" the data :

```R
library(lidR)
library(sf)

x <- "dir/DONNE_NON_CLASSIFIEE/"

ZI_LAS <- readLAScatalog(x)  # Las file from IGN (Test urban area of Manosque (04) in France) 

opt_chunk_buffer(ZI_LAS) <- 20
opt_chunk_size(ZI_LAS)   <- 250
opt_laz_compression(ZI_LAS) <- T
opt_output_files(ZI_LAS) <- paste0("dir/declass_{XCENTER}_{YCENTER}_{ID}")

box <- extent(ZI_LAS)

classify_poi(ZI_LAS, class = as.integer(1), roi = box)

```
**R 4.0.3**
Package Version : 
`sf`= 1.0.7
`lidR` = 4.0.0 
`raster` = 3.5.15
`knitr` = 1.40
`shiny` = 1.7.2
`manipulateWidget` = 0.11.1
`ggplot2` = 3.3.6

Here you can see my entire script that I use (reduce to a minimum) : 

```R
library(lidR)
library(knitr)
library(shiny)
library(manipulateWidget)
library(raster)
library(ggplot2)
library(sf)  # Linking to GEOS 3.9.1, GDAL 3.2.1, PROJ 7.2.1; sf_use_s2() is TRUE

set_lidr_threads(14)

dir_bat <- choose.files(default = "", caption = "Selectionner le fichier shapefile des bâtiments de votre zone d'interet :",
                          multi = FALSE, filters = Filters["All",])

BAT <- st_read(dir_bat, quiet = T)

ZI_LAS <- readLAScatalog("dir/path to unclassified data/")

ZI_SOL <- Reclass_las_csf(ZI_LAS,cfs_Thr,cfs_Res,cfs_Rig)

cfs_Thr = 0.05
cfs_Res = 0.5
cfs_Rig = 1

Reclass_las_csf <- function(ZI_LAS, cfs_Thr, cfs_Res, cfs_Rig ){
  
  opt_chunk_buffer(ZI_LAS) <- 20
  opt_chunk_size(ZI_LAS)   <- 250
  plot(ZI_LAS, chunk = TRUE)
  
  
  mycfs <- csf(
    sloop_smooth      <- FALSE,
    class_threshold   <- cfs_Thr,
    cloth_resolution  <- cfs_Res,
    rigidness         <- cfs_Rig,
    iterations        <- 500,
    time_step         <- 0.65
  ) 
  
  dir.create(paste0("SOL/",cfs_Thr,"_",cfs_Res,"_",cfs_Rig))
  dir_sol <- paste0("SOL/",cfs_Thr,"_",cfs_Res,"_",cfs_Rig)
  
  opt_laz_compression(ZI_LAS) <- FALSE
  opt_output_files(ZI_LAS) <- paste0(dir_sol,"/Sol_{XCENTER}_{YCENTER}_{ID}")
  ZI_SOL <- classify_ground(ZI_LAS,algorithm = mycfs)
    
  dir.create(paste0(dir_sol,"/BAT"))
  dir_BAT = paste0(dir_sol,"/BAT")
  
  opt_output_files(ZI_SOL) <- paste0(dir_BAT,"/Bat_{XCENTER}_{YCENTER}_{ID}")
  st_crs(BAT) <- st_crs(ZI_SOL)
  
  ZI_BAT <- classify_poi(ZI_SOL, class = as.integer(6), roi = BAT )
  
  ZI_SOL <- readLAScatalog(dir_bat)
  
  return(ZI_SOL)
}
```

![Anim_LidR_PB](https://user-images.githubusercontent.com/116190503/196957372-7b8f9a05-bd35-4b9f-b35d-8ef5a6317e1d.gif)

**Reply by Jean-Romain** [reply]
This is not a minimal reproducible example

**Reply by MaximeAuffret** [reply]
What do you mean by minimal ? 

is that minimal ?  :
```R
library(lidR)
library(sf)

list = list.files(pattern = ".las")
LAS = readLAScatalog(list)
bat = st_read("BAT.shp")

opt_output_files(LAS) = "dir/"

classify_poi(LAS, class = as.integer(6), roi = bat)
```
I still have the error with this script.

**Reply by Jean-Romain** [reply]
Almost mimaly reproducible but I can't copy paste it to reproduce and I need to make modifications to make it working. The following seems minimal but not reproducible because it works for me. Can you confirm if it works for you?

I'm using 4.0.3 but I don't remember I changed anything in `classify_poi()` since 4.0.0. There is nothing in the changelog.

```r
library(lidR)
library(sf)

LAS = readLAScatalog("issue 630/DONNEES_NON_CLASSIFIES/")
bat = st_read("issue 630/BAT/BAT.shp")

opt_output_files(LAS) = paste0(tempdir(),"/*")

st_crs(bat) <- st_crs(LAS)
classify_poi(LAS, class = as.integer(6), roi = bat)
```

**Reply by MaximeAuffret** [reply]
It is not working. 
I am re-installing Rtools 4 and updating all the packages again to see.

**Reply by Jean-Romain** [reply]
You can only install 4.0.2 from CRAN

**Reply by MaximeAuffret** [reply]
It seems to be working now. I don't know what happened but it was certainly the packages. 
Maybe the version of `Rtools` ?

Now I've got : 
`sf`= 1.0.8 (1.07 before)
`lidR` = 4.0.1 (4.0.0 before)

Thank you for your help. And I apologize for the inconvenience.

**Reply by Jean-Romain** [reply]
Ok, I don't know either. I'm glad it works
